### PR TITLE
CERNAccess: Request CERN access from a registration form

### LINKED ISSUE #21: CERN Access: "personal data" reg. form section
Let's add a "personal data" section within the registration form (in events where CERN Access is active). Users who fill it in will be "accepted" by default.

### PR DESCRIPTION
**Pull request by mic4ael**
closes #21

### CODE REVIEW

File: cern_access/indico_cern_access/templates/regform_identity_data_section.html [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+                registrationForm.find('.regform-section-list').append(personalDataForm);
+            });
+
+            $(document).ready(function() {
+                var personalDataForm = $('.personal-data-form').find('.i-box');
+
+                personalDataForm.hide();
+                $('#toggle-personal-data-form').on('change', function() {
+                    personalDataForm.toggle();
```
Comment (ThiefMaster) [inline cern_access/indico_cern_access/templates/regform_identity_data_section.html]: that's just asking for things to get out of sync. it's better to toggle it to the `this.checked` status and then trigger the change event handler afterwards to have the correct state before the first real change on the field.

File: cern_access/indico_cern_access/plugin.py [hunk 2]
```diff
@@ -122,8 +123,11 @@ def init(self):
         super(CERNAccessPlugin, self).init()
         self.template_hook('registration-status-flag', self._get_access_status)
         self.template_hook('registration-status-action-button', self._get_access_action_button)
+        self.template_hook('additional-form-sections', self._get_personal_data_form)
         self.connect(signals.plugin.get_event_request_definitions, self._get_event_request_definitions)
         self.connect(signals.event.registration_deleted, self._registration_deleted)
+        self.connect(signals.event.registration_created, self._registration_action)
```
Comment (ThiefMaster) [inline cern_access/indico_cern_access/plugin.py]: `_registration_created`

File: cern_access/indico_cern_access/util.py [hunk 3]
```diff
@@ -285,7 +285,11 @@ def grant_access(registrations, regform, email_subject, email_body, email_sender
                                  reg.cern_access_request.request_state == CERNAccessRequestState.active)]
     state, data = send_adams_post_request(event, new_registrations)
     add_access_requests(new_registrations, data, state)
-    send_form_link(new_registrations, email_subject, email_body, email_sender)
+    for registration in [r for r in new_registrations if r.cern_access_request.has_identity_info]:
```
Comment (ThiefMaster) [inline cern_access/indico_cern_access/util.py]: IMHO this is super ugly :) I'd rather do an `if...: continue` inside the body

File: cern_access/indico_cern_access/templates/regform_identity_data_section.html [hunk 4]
```diff
@@ -0,0 +1,69 @@
+{% from 'cern_access:_common.html' import render_access_dates %}
+{% from 'forms/_form.html' import form_rows, form_row %}
+{% from 'message_box.html' import message_box %}
```
Comment (ThiefMaster) [inline cern_access/indico_cern_access/templates/regform_identity_data_section.html]: unused

File: cern_access/indico_cern_access/templates/grant_access.html [hunk 5]
```diff
@@ -11,6 +11,11 @@
         {% trans -%}
             Site access will be granted automatically once they have provided the requested data.
         {%- endtrans %}
+        <br>
+        {% trans -%}
+            Remember that those users who already provided the requested data will immediately receive
+            the message with the ticket.
```
Comment (ThiefMaster) [inline cern_access/indico_cern_access/templates/grant_access.html]: this is not clear, it sounds like they still get the message you entered

File: cern_access/indico_cern_access/plugin.py [hunk 6]
```diff
@@ -153,6 +157,16 @@ def _get_access_status(self, regform, registration, header, **kwargs):
                                           registration=registration,
                                           header=header)
 
+    def _get_personal_data_form(self, event, regform, management, registration=None, **kwargs):
+        if management or registration is not None:
+            return
+
+        form = RegistrationFormPersonalDataForm()
```
Comment (ThiefMaster) [inline cern_access/indico_cern_access/plugin.py]: If you use `form = g.get('personal_data_form') or RegistrationFormPersonalDataForm()` here you should get proper form error display without having to flash them manually

File: cern_access/indico_cern_access/plugin.py [hunk 7]
```diff
@@ -163,6 +177,37 @@ def _registration_deleted(self, registration, **kwargs):
...
+        personal_data_form = g.pop('personal_data_form', None)
+
+        if not regform.cern_access_request or not regform.cern_access_request.is_active or not personal_data_form:
+            return
+
+        personal_data = personal_data_form.data
+        if not personal_data['request_cern_access']:
+            return
+
+        request = registration.cern_access_request or CERNAccessRequest()
```
[the remark this answers is not in the document]
Comment (mic4ael) [reply cern_access/indico_cern_access/plugin.py]: i think it is a relic of what I did before and eventually removed

**Pull request merged**